src/compiler.ts

```diff
@@ -6622,6 +6632,33 @@ export class Compiler extends DiagnosticEmitter {
     return module.block(null, exprs);
   }
 
+  private writebackClosedLocals(
+    closureContextLocal: Local
+  ): ExpressionRef {
+    var module = this.module;
+    var type = closureContextLocal.type;
+    var locals = assert(type.locals);
+    let _values = Map_values(locals);
+    var exprs = new Array<ExpressionRef>(_values.length);
+    assert(type.is(TypeFlags.REFERENCE));
+    for (let i = 0, k = _values.length; i < k; ++i) {
+      let local = unchecked(_values[i]);
+      let nativeType = local.type.toNativeType();
+      assert(local.closureContextOffset > 0);
+      exprs[i] = module.local_set(
+        local.index,
+        module.load(
+          local.type.byteSize,
+          local.type.is(TypeFlags.SIGNED),
+          module.local_get(closureContextLocal.index, this.options.nativeSizeType),
+          nativeType,
+          local.closureContextOffset
+        )
+      );
+    }
+    return module.block(null, exprs);
+  }
```

Comment:
Just the opposite of `injectClosedLocals` where we put the values in closureContextLocal back into the locals of the calling function